src/basic/macro.c

```diff
@@ -0,0 +1,65 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
```

Comment:
hmm, it's a bit weird having a .c file that is called macro.c...

maybe name this like the contents hence, not after the concepts used to implement them. hence maybe assert-util.[ch]?

Reply:
This would introduce circular dependencies again because these functions use macros from macro.h. I'm pretty sure if we want to untangle this it should be a separate PR.

Reply:
NVM the prev comment, fixed